src/material/schematics/ng-generate/mdc-migration/rules/style-migrator.ts

```diff
@@ -107,4 +112,37 @@ export abstract class StyleMigrator {
       }
     }
   }
+
+  /**
+   * Returns whether the given postcss rule uses a potentially deprecated
+   * selector of the old component.
+   *
+   * @param rule a postcss rule.
+   * @returns `true` if the given Rule uses a selector with the deprecated prefix.
+   */
+  isDeprecatedSelector(rule: postcss.Rule): boolean {
+    return rule.selector.includes(this.deprecatedPrefix);
+  }
+
+  /**
+   * Adds comment before declaration that says the following rule may not apply
+   * in the MDC version for that component
+   *
+   * @param rule a postcss rule.
+   */
+
+  addDeprecatedSelectorComment(rule: postcss.Rule): void {
+    let comment = postcss.comment({
+      text:
+        'TODO: The following rule targets internal classes of ' +
+        this.component +
+        ' that may no longer apply for the MDC version.',
+    });
+    // We need to manually adjust the indentation and add new lines between the
+    // comment and declaration
+    const indentation = rule.raws.before?.split('\n').pop();
+    comment.raws.before = '\n' + indentation;
+    rule.parent?.insertBefore(rule, comment);
```

Comment:
Nit: What are the cases where `parent` is undefined? I assume we'd never hit this right? if so, maybe use `!`/and add a comment? not feeling strongly though